### PR TITLE
bugfix for the mseed gaps

### PR DESCRIPTION
**Pull request by haruming**
This is the bugfix for issue #261 
Basically, we use the obspy built-in merge function to fix this.

### CODE REVIEW

File: python/mspasspy/db/database.py [hunk 1]
```diff
@@ -203,8 +203,21 @@ def set_database_schema(self, schema):
...
+        normalize=None,
+        load_history=False,
+        exclude_keys=None,
+        collection="wf",
+        data_tag=None,
+        alg_name="read_data",
+        alg_id="0",
+        define_as_raw=False,
+        retrieve_history_record=False,
+        fill_value=None,
```
Comment (wangyinz) [inline python/mspasspy/db/database.py:219]: We should probably add all three parameters from obspy here: `Stream.merge(method=0, fill_value=None, interpolation_samples=0)`. We can point user to obspy's detailed documentation here, which includes [this](https://docs.obspy.org/packages/autogen/obspy.core.trace.Trace.__add__.html#handling-overlaps) and [that](https://docs.obspy.org/packages/autogen/obspy.core.stream.Stream._cleanup.html#obspy.core.stream.Stream._cleanup).

File: python/mspasspy/db/database.py [hunk 2]
```diff
@@ -269,6 +282,16 @@ def read_data(self, object_id, mode='promiscuous', normalize=None, load_history=
...
         :type data_tag: :class:`str`
+        :param alg_name: alg_name is the name the func we are gonna save while preserving the history.
+        :type alg_name: :class:`str`
+        :param alg_id: alg_id is a unique id to record the usage of func while preserving the history.
+        :type alg_id: :class:`bson.objectid.ObjectId`
+        :param define_as_raw: a boolean control whether we would like to set_as_origin when loading processing history
+        :type define_as_raw: :class:`bool`
+        :param retrieve_history_record: a boolean control whether we would like to load processing history
+        :type retrieve_history_record: :class:`bool`
+        :param fill_value: Fill value for gaps. Defaults to None. Traces will be converted to NumPy masked arrays if no value is given and gaps are present.
```
Comment (wangyinz) [inline python/mspasspy/db/database.py:293]: "Traces will be converted to NumPy masked arrays if no value is given and gaps are present" I don't think this is true. This is what obspy does, but in our data object we don't have such feature.

**Review by wangyinz — CHANGES_REQUESTED** [review]
Very good. There is only some small issues need to be addressed.

File: python/mspasspy/db/database.py (hunk 1 above)
Comment (haruming) [reply python/mspasspy/db/database.py:219]: Agreed.

File: python/mspasspy/db/database.py [hunk 3]
```diff
@@ -216,7 +216,9 @@ def read_data(
         alg_id="0",
         define_as_raw=False,
         retrieve_history_record=False,
+        method=0,
```
Comment (wangyinz) [inline python/mspasspy/db/database.py:219]: Sorry that I should've thought about this: we should probably name it "merge_method" instead. Probably should also add "merge_" to the next two parameters. The names are confusing as this is the read_data method.

**Review by wangyinz — CHANGES_REQUESTED** [review]

File: python/mspasspy/db/database.py (hunk 3 above)
Comment (haruming) [reply python/mspasspy/db/database.py:219]: Yeah I agree that it's a bit confusing and change to `merge_` it makes sense.

**Comment by wangyinz** [comment]
hmmm.... why did it failed? Could you please look into it @haruming ? Thank you!

**Pull request merged**